modules/programs/atuin.nix

```
@@ -0,0 +1,157 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.atuin;
+
+  tomlFormat = pkgs.formats.toml { };
+
+in {
+  meta.maintainers = [ maintainers.hawkw ];
+
+  options.programs.atuin = {
+    enable = mkEnableOption "atuin";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.atuin;
+      defaultText = literalExample "pkgs.atuin";
+      description = "The package to use for atuin.";
+    };
+
+    enableBashIntegration = mkEnableOption "Bash integration" // {
+      default = true;
+      description = ''
+        Whether to enable Atuin's Bash integration.
+        </para><para>
+        If enabled, this will bind <literal>ctrl-r</literal> to open the Atuin
+        history.
+      '';
+    };
```

Can make it shorter by simply using `mkOption`, 
```suggestion
    enableBashIntegration = mkOption {
      type = types.bool;
      default = true;
      description = ''
        Whether to enable Atuin's Bash integration. This will bind
        <literal>ctrl-r</literal> to open the Atuin history.
      '';
    };
```